### PR TITLE
fix(source-google-sheets): fail the sync attempt when google responds with an api error

### PR DESCRIPTION
**Pull request by Reidsy**
## What

- When google sheets responds with an api error, airbyte ignores this and completes the sync with zero rows of data effectively emptying the destination table
- Related Issue: https://github.com/airbytehq/airbyte/issues/45106

## How

- Raises an AirbyteTracedException and stops syncing when a google sheets api error is encountered

## Review guide

1. `airbyte-integrations/connectors/source-google-sheets/source_google_sheets/source.py`
2. `airbyte-integrations/connectors/source-google-sheets/unit_tests/test_stream.py`

This is a very similar fix to https://github.com/airbytehq/airbyte/pull/41993 for a similar issue.
I think I would keep the handling of the issues separate as one is related to rate limiting, wheres this handles any unidentified error, but let me know if there is a reason to combine the two blocks of code.

## User Impact

- No more syncs that fail silently resulting in an empty table

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Review by eenu5 — APPROVED** [review]

File: airbyte-integrations/connectors/source-google-sheets/source_google_sheets/source.py [hunk 1]
```diff
@@ -259,6 +259,11 @@ def read(
                 ) from e
             else:
                 logger.info(f"{e.status_code}: {e.reason}. {error_description}")
+                raise AirbyteTracedException(
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-google-sheets/source_google_sheets/source.py:262]: @Reidsy, can you update the connector version in the metadata.yaml and pyproject.toml files? Additionally, please add an entry in the changelog connector docs (docs/integrations/source/...).

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-google-sheets/source_google_sheets/source.py (hunk 1 above)
Comment (Reidsy) [reply airbyte-integrations/connectors/source-google-sheets/source_google_sheets/source.py:262]: Bumped version to `0.7.4` and added information to changelog

**Comment by Reidsy** [comment]
Is there anything else you need me to do here? Looks like it's approved and ready to merge

**Comment by natikgadzhi** [comment]
Kicked off full CI.

**Review by natikgadzhi — APPROVED** [review]
The code looks good to me. Let's wait for the tests to pass.

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**